src/test/test_vcfreader.cpp

```diff
@@ -0,0 +1,192 @@
+        command="rm -f tmp1_vcf"; run_command(command);
+        command="bcftools annotate --threads "+N_THREADS+" -x INFO/AC,INFO/AF,INFO/AN "+tmp2_vcf.string();
+        run_command(command,tmp1_vcf);
+        command="rm -f "+tmp2_vcf.string(); run_command(command);
+        command="bcftools +fill-tags "+tmp1_vcf.string()+" -Ob -o "+input_vcf.string()+" -- -t AC,AF,AN,F_MISSING"; run_command(command);
+        command="rm -f tmp1_vcf"; run_command(command);
+        // Filtering by QUAL
+        high_qual_only=true; pass_only=false; min_sv_length=0; skip_samples=false; min_af=0.0; min_nmf=0.0;
+        for (auto& min_qual: MIN_QUALS) {
+            command="bcftools filter --threads "+N_THREADS+R"( --include "QUAL>=)"+to_string(min_qual)+R"(" )"+input_vcf.string()+R"( | grep "^[^#]" | sed 's/PASS/./g')";
+            run_command(command,truth_vcf);
+            test("HPRC",chromosome,is_diploid,high_qual_only,min_qual,pass_only,min_sv_length,skip_samples,n_samples_in_vcf,min_af,min_nmf);
+        }
+        // Filtering by FILTER skipped, since FILTER is always '.'
+        // Filtering by SVLEN skipped, since the SVLEN tag is missing, bcftools does not work for adding it, and
+        // `truvari anno svinfo` can add it but alters the VCF in other ways that break the test.
+        // Removing all sample columns
+        high_qual_only=false; min_qual=0.0; pass_only=false; min_sv_length=0; skip_samples=true; min_af=0.0; min_nmf=0.0;
+        command="cut -f 1-9 "+input_vcf.string()+R"( | grep "^[^#]")";
+        run_command(command,truth_vcf);
+        test("HPRC",chromosome,is_diploid,high_qual_only,min_qual,pass_only,min_sv_length,skip_samples,n_samples_in_vcf,min_af,min_nmf);
+        // Filtering by AF
+        high_qual_only=false; min_qual=0.0; pass_only=false; min_sv_length=0; skip_samples=false; min_nmf=0.0;
+        for (auto& min_af: MIN_AFS) {
+            command="bcftools filter --threads "+N_THREADS+R"( --include "AC/)"+to_string(n_haplotypes)+">="+to_string(min_af)+R"(" )"+input_vcf.string()+R"( | grep "^[^#]" | sed 's/PASS/./g')";
+            run_command(command,truth_vcf);
+            test("HPRC",chromosome,is_diploid,high_qual_only,min_qual,pass_only,min_sv_length,skip_samples,n_samples_in_vcf,min_af,min_nmf);
+        }
+        // Filtering by MIN_NMF
+        high_qual_only=false; min_qual=0.0; pass_only=false; min_sv_length=0; skip_samples=false; min_af=0.0;
+        for (auto& min_nmf: MIN_NMFS) {
+            command="bcftools filter --threads "+N_THREADS+R"( --include "AN/)"+to_string(n_haplotypes)+">="+to_string(min_nmf)+R"(" )"+input_vcf.string()+R"( | grep "^[^#]" | sed 's/PASS/./g')";
+            run_command(command,truth_vcf);
+            test("HPRC",chromosome,is_diploid,high_qual_only,min_qual,pass_only,min_sv_length,skip_samples,n_samples_in_vcf,min_af,min_nmf);
+        }
+    }
+}
+
+/**
+ * Requirements:
+ * - the environment variable `BCFTOOLS_PLUGINS` should be set.
```

Comment:
Do we need to add this to the list of dependencies for the project?

Reply:
Those plugins are shipped with bcftools. The only thing that is not set by bcftools' installation is the environment variable, which I wasn't able to set and export inside the C++ program.